invert-gradle-plugin/src/main/kotlin/com/squareup/invert/internal/report/sarif/InvertSarifReportWriter.kt

```diff
@@ -0,0 +1,113 @@
+package com.squareup.invert.internal.report.sarif
+
+import com.squareup.invert.internal.InvertFileUtils
+import com.squareup.invert.internal.models.InvertPluginFileKey
+import com.squareup.invert.logging.InvertLogger
+import com.squareup.invert.models.Stat
+import com.squareup.invert.models.StatMetadata
+import com.squareup.invert.models.js.StatsJsReportModel
+import io.github.detekt.sarif4k.ReportingDescriptor
+import io.github.detekt.sarif4k.Result as SarifResult
+import io.github.detekt.sarif4k.Run
+import io.github.detekt.sarif4k.SarifSchema210
+import io.github.detekt.sarif4k.SarifSerializer
+import io.github.detekt.sarif4k.Tool
+import io.github.detekt.sarif4k.ToolComponent
+import io.github.detekt.sarif4k.Version
+import java.io.File
+import java.nio.file.Files
+
+/**
+ * Writer for generating SARIF (Static Analysis Results Interchange Format) reports for Invert.
+ * SARIF is a standard format for static analysis tools to report their results.
+ */
+class InvertSarifReportWriter(
+    private val logger: InvertLogger,
+    rootBuildReportsDir: File,
+) {
+    private val rootBuildSarifReportsDir = File(rootBuildReportsDir, InvertFileUtils.SARIF_FOLDER_NAME)
+
+    /**
+     * Writes a SARIF report to a file.
+     */
+    private fun writeToSarif(
+        jsonFileKey: InvertPluginFileKey,
+        rulesAndResults: Map<ReportingDescriptor, List<SarifResult>>
+    ) {
+        val sarif = createSarifSchema(
+            toolName = "Invert",
+            toolVersion = "1.0.0",
+            rule = rulesAndResults.keys.toList(),
+            results = rulesAndResults.values.flatten()
+        )
+
```

Comment:
Nit spacing. We definitely need to add killing/spotless to the repo. The time has come.